### PR TITLE
doc(python): Improve Python SDK docs / added releases and session documentation.

### PR DESCRIPTION
**Pull request by antonpirker**
- cleanup: removed ancient "new in version 0.x" mentions
- reorganization: reordered subitems of "configuration" section in python guides to the actual order a new user configures/discovers the different features
- updated docs on releases and sessions so that it is more clear what this is.

### CODE REVIEW

File: src/includes/configuration/auto-session-tracking/python.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+We create a session for every request and response cycle.
+Web socket connections currently do not create sessions.
+
+We mark the session as:
+
+- `crashed` if an _unhandled error_ is raised.
+- `an error` if the SDK captures an event that contains an exception (this includes manually captured errors).
+
+By default, the Python SDK is sending sessions. To disable sending sessions, set the `auto_session_tracking` flag to `False`:
```
Comment (imatwawana) [inline src/includes/configuration/auto-session-tracking/python.mdx:9]:
```suggestion
By default, the Python SDK sends sessions. To disable sending sessions, set the `auto_session_tracking` flag to `False`:
```

File: src/platforms/common/configuration/releases.mdx [hunk 2]
```diff
@@ -46,26 +48,38 @@ The value can be arbitrary, but we recommend either of these naming strategies:
...
 After configuring your SDK, you can install a repository integration or manually supply Sentry with your own commit metadata. Read our documentation about [setting up releases](/product/releases/setup/) for further information about integrations, associating commits, and telling Sentry when deploying releases.
 
 <PlatformSection supported={["apple", "android", "javascript", "flutter", "native", "rust", "python", "react-native", "dotnet", "unity"]}>
 
 ## Release Health
-  
+
+### Sessions
+
+A session represents the interaction between the user and the application. Sessions contain a timestamp, a duration, a status (if the session was OK or if it crashed) and others.
```
Comment (imatwawana) [inline src/platforms/common/configuration/releases.mdx:75]: Ending that second sentence with "and others" leaves it open for the question "what else?". If we're not prepared to answer that, then maybe we should just end the list at status. If you're dead set on keeping that, then I've made an edit to fix the wording. 
 
```suggestion
A session represents the interaction between the user and the application. Sessions contain a timestamp, a duration, a status (if the session was OK or if it crashed), and other information.
```

File: src/platforms/common/configuration/releases.mdx [hunk 3]
```diff
@@ -46,26 +48,38 @@ The value can be arbitrary, but we recommend either of these naming strategies:
...
+
+Sessions are always linked to a release and are used to monitor user adoption, usage of the application, percentage of [crashes](/product/releases/health/#crash), and [session data](/product/releases/health/#session) of that release.
+
+This will give you insight into the impact of crashes and bugs as it relates to user experience, and reveal trends with each new issue through the release details, graphs, and filters.
```
Comment (imatwawana) [inline src/platforms/common/configuration/releases.mdx:79]: Connecting these two thoughts into one paragraph or the second part becomes a bit confusing in terms of what "this" is referring to. 

Which graphs and filters are you specifically talking about here? The ones on the Release Details page?

```suggestion
Sessions are always linked to a release and are used to monitor user adoption, usage of the application, percentage of [crashes](/product/releases/health/#crash), and [session data](/product/releases/health/#session) of that release. This will give you insight into the impact of crashes and bugs on user experience and reveal trends with each new issue through the release details, graphs, and filters.
```

File: src/platforms/python/common/configuration/integrations/index.mdx [hunk 4]
```diff
@@ -1,10 +1,34 @@
...
+sentry_sdk.init(
+    ...
+    integrations=[
+        DjangoIntegration(),
+        CeleryIntegration(),
+        RedisIntegration(),
+    ],
+)
+```
+
```
Comment (imatwawana) [inline src/platforms/python/common/configuration/integrations/index.mdx:24]: This appears to duplicate similar content that already exists on the child pages of this page and I don't think it needs to be repeated. Take a look at the JavaScript Integrations page for an example of what this page generally looks like. I think the prose is that you've added is fine, but the code snippet seems out of place.

File: src/platforms/python/common/configuration/integrations/index.mdx [hunk 5]
```diff
@@ -1,10 +1,34 @@
...
+**NEW:** The Sentry Python SDK also tries to automatically enable integrations that fit your Python modules you have installed. If you have Django installed the Sentry SDK will enable the `DjangoIntegration()` by default for you. No need for configuration at all!
```
Comment (imatwawana) [inline src/platforms/python/common/configuration/integrations/index.mdx:25]: Adding markers like "new" on content means you then have to think about keeping track of when it's no longer new. Err on the side of making the content future-proof. Also, always avoid exclamation points. Generally we go for an informative, but casual tone.

```suggestion
The Sentry Python SDK also tries to automatically enable integrations that fit the Python modules you have installed. If you have Django installed, the Sentry SDK will enable the `DjangoIntegration()` by default for you without any further configuration.
```

File: src/platforms/python/common/configuration/integrations/index.mdx [hunk 6]
```diff
@@ -1,10 +1,34 @@
...
+
+If you do not want this behavior, set the `default_integrations` parameter to `False`. (See [Integration Configuration](/platforms/python/configuration/options/#integration-configuration) on the Basic Options page)
```
Comment (imatwawana) [inline src/platforms/python/common/configuration/integrations/index.mdx:27]: Unless there's a good reason not to, we generally use contractions. 

```suggestion
If you don't want integrations to be enabled automatically, set the `default_integrations` parameter to `False`. Learn more about <PlatformLink to="/configuration/options/#integration-configuration">integration options</PlatformLink>integration configuration.
```

**Review by imatwawana — COMMENTED** [review]
This is a really excellent start! Most of my fixes are around style, but there are a couple significant structural questions that I have. This should definitely be reviewed by another developer who's worked on the docs before as well specifically to review changes you've made to the common content, and sidebar order change to the configuration topics as those changes affect all SDKs, as well as the "new in version" references. After that I'd like to do another pass.

File: src/platforms/common/configuration/releases.mdx [hunk 7]
```diff
@@ -77,10 +91,4 @@ Support for server frameworks such as ASP.NET is planned.
 
 </PlatformSection>
 
-Monitor the [health of releases](/product/releases/health/) by observing user adoption, usage of the application, percentage of [crashes](/product/releases/health/#crash), and [session data](/product/releases/health/#session). Release health will provide insight into the impact of crashes and bugs as it relates to user experience, and reveal trends with each new issue through the release details, graphs, and filters.
-
-The SDK will automatically manage the start and end of the sessions when the SDK is initialized.
-
-<PlatformContent includePath="configuration/auto-session-tracking" />
-
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply src/platforms/common/configuration/releases.mdx:85]: I have brought that paragraph back.

File: src/platforms/python/common/configuration/integrations/index.mdx (hunk 4 above)
Comment (antonpirker) [reply src/platforms/python/common/configuration/integrations/index.mdx:24]: It is a similar content as in the child pages, but it is illustrating, that one can add multiple integrations (which is not shown elsewhere) and also it enables Django, Celery and Redis which are possibly the most used combinations for python web projects and there it can be kind of a shortcut for users doing a lot of Django projects. They just need to copy/paste this and add the dsn and they are done

**Comment by antonpirker** [comment]
I have now fixed almost all your comments.

File: src/platforms/common/configuration/releases.mdx [hunk 8]
```diff
@@ -32,12 +32,12 @@ Additionally, releases are used for applying [source maps](/platforms/javascript
 
 ## Bind the Version
 
-Include a release ID (often called a "version") when you configure your client SDK.
+Include a release ID (often called a "version") when you initialise the SDK.
```
Comment (bruno-garcia) [inline src/platforms/common/configuration/releases.mdx:35]: We prefer the US spelling on the docs for consistency:

```suggestion
Include a release ID (often called a "version") when you initialize the SDK.
```

File: src/platforms/common/configuration/releases.mdx [hunk 9]
```diff
@@ -46,26 +46,44 @@ The value can be arbitrary, but we recommend either of these naming strategies:
...
+
+In order to monitor release health the SDK sends session data.
+
+### Sessions
+
+A session represents the interaction between the user and the application. Sessions contain a timestamp, a duration, a status (if the session was OK or if it crashed), and other information.
+
+Sessions are always linked to a release.
+
+The SDK will automatically manage the start and end of the sessions when the SDK is initialized.
```
Comment (bruno-garcia) [inline src/platforms/common/configuration/releases.mdx:83]: I realize you just moved the content but since not all SDKs can start/end sessions automatically and the way they work varies a bit, maybe reword somehow to be more generic?

```suggestion
Most Sentry SDKs can manage sessions automatically. For example, mobile SDKs can end session when the app goes to the background for longer than 30 seconds. Other SDKs end the session when the process exit.
```

**Review by bruno-garcia — COMMENTED** [review]
I know it's a draft but I wanted to leave a note since it changes the common content of releases and affects all platforms

**Comment by imatwawana** [comment]
> To the change in order of the subpages of "configuration": They are now in the "natural" order that someone implements Sentry for the first time. First your set the `basic options` and play around. Then you (maybe) enable a `integration`. If you play around longer you will set` environments` and then your will configure `releases`. If you run into problems because your site is to famous to add `sampling` and/or `filtering`. If you somehow miss events you configure `draining`.
> 
> But it is very welcome so ask another developer if she/he agrees with this ordering.

I got some input on this and the order is good!

File: src/platforms/python/common/configuration/integrations/index.mdx (hunk 4 above)
Comment (imatwawana) [reply src/platforms/python/common/configuration/integrations/index.mdx:24]: @untitaker, your opinion on this would be helpful here. I'm concerned about this Integrations page being very different from other ones. Do you think this information is important enough to keep on this page vs (perhaps) having it as an extra code snippet on the child pages?

File: src/platforms/common/configuration/releases.mdx (hunk 9 above)
Comment (antonpirker) [reply src/platforms/common/configuration/releases.mdx:83]: I also feed the original sentence a little bit confusing. Maybe we just need to tell people "The SDK will automatically send session information" So they know that sessions are recorded, when exactly they start and stop is imho not that important here. What do you think?

File: src/platforms/python/common/configuration/integrations/index.mdx (hunk 4 above)
Comment (antonpirker) [reply src/platforms/python/common/configuration/integrations/index.mdx:24]: The main thing why I put this here is that developers are lazy and don't want to think about integrating third party libraries. If your are doing a new Django project you can just go to "Configuration" -> "Integrations" then copy this snipped and you are done. 
But because we have also automatic enabling of integrations this is not necessary at all. (we do not have documented automatic enabling of integrations anywhere, but this is another discussion for another place)

So I am also fine with removing it.

Comment (antonpirker) [reply src/platforms/python/common/configuration/integrations/index.mdx]: I have now removed the code snippet. After reading the docs guide and thinking about it, it is really a duplicated code snippet.

File: src/platforms/common/configuration/releases.mdx (hunk 9 above)
Comment (antonpirker) [reply src/platforms/common/configuration/releases.mdx]: I have now applied your suggestion, but removed the examples, because a few paragraphs below is described what a session is. I think the description of what is a session, should be done in the platform specific sections.

Comment (imatwawana) [reply src/platforms/common/configuration/releases.mdx]: @bruno-garcia - can you take another look, please?

File: src/includes/configuration/auto-session-tracking/python.mdx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+We create a session for every request and response cycle.
+
+Web socket connections currently do not create sessions.
+
+We mark the session as:
+
+- `crashed` if an _unhandled error_ is raised.
+- `an error` if the SDK captures an event that contains an exception (this includes manually captured errors).
```
Comment (imatwawana) [inline src/includes/configuration/auto-session-tracking/python.mdx:8]: Trying to improve the flow here.

```suggestion
We create a session for every request and response cycle and mark the session as:

- `crashed` if an _unhandled error_ is raised.
- `an error` if the SDK captures an event that contains an exception (this includes manually captured errors).

Currently, web socket connections don't create sessions.
```

File: src/includes/configuration/auto-session-tracking/ruby.mdx [hunk 11]
```diff
@@ -0,0 +1,16 @@
+We create a session for every request and response cycle.
+
+Web socket connections currently do not create sessions.
+
+We mark the session as:
+
+- `crashed` if an _unhandled error_ is raised.
+- `an error` if the SDK captures an event that contains an exception (this includes manually captured errors).
```
Comment (imatwawana) [inline src/includes/configuration/auto-session-tracking/ruby.mdx:8]: Same as above

```suggestion
We create a session for every request and response cycle and mark the session as:

- `crashed` if an _unhandled error_ is raised.
- `an error` if the SDK captures an event that contains an exception (this includes manually captured errors).

Currently, web socket connections don't create sessions.
```

File: src/platforms/common/configuration/environments.mdx [hunk 12]
```diff
@@ -1,12 +1,16 @@
...
+sidebar_order: 30
 description: "Learn how to configure your SDK to tell Sentry about your environments."
 notSupported:
   - perl
 ---
 
-Sentry automatically creates environments when it receives an event with the environment tag. Environments are case sensitive. The environment name can't contain newlines, spaces or forward slashes, can't be the string "None", or exceed 64 characters. You can't delete environments, but you can [hide](/product/sentry-basics/environments/#hidden-environments) them.
+Sometimes you want to know if an error was received from the production system or the staging server. This is what environments are for.
+
+Sentry automatically creates environments when it receives an event with the `environment` tag set.
```
Comment (imatwawana) [inline src/platforms/common/configuration/environments.mdx:11]:
```suggestion
Sentry automatically creates an environment when it receives an event with the `environment` tag set.
```

File: src/platforms/common/configuration/environments.mdx [hunk 13]
```diff
@@ -1,12 +1,16 @@
...
 title: Environments
-sidebar_order: 5
+sidebar_order: 30
 description: "Learn how to configure your SDK to tell Sentry about your environments."
 notSupported:
   - perl
 ---
 
-Sentry automatically creates environments when it receives an event with the environment tag. Environments are case sensitive. The environment name can't contain newlines, spaces or forward slashes, can't be the string "None", or exceed 64 characters. You can't delete environments, but you can [hide](/product/sentry-basics/environments/#hidden-environments) them.
+Sometimes you want to know if an error was received from the production system or the staging server. This is what environments are for.
```
Comment (imatwawana) [inline src/platforms/common/configuration/environments.mdx:9]:
```suggestion
Environments tell you where an error occurred, whether that's in your production system, your staging server, or elsewhere. 
```

File: src/platforms/common/configuration/releases.mdx [hunk 14]
```diff
@@ -46,26 +46,44 @@ The value can be arbitrary, but we recommend either of these naming strategies:
...
+If you don't tell Sentry about a new release, Sentry will automatically create a release entity in the system the first time it sees an event with that release ID.
 
 After configuring your SDK, you can install a repository integration or manually supply Sentry with your own commit metadata. Read our documentation about [setting up releases](/product/releases/setup/) for further information about integrations, associating commits, and telling Sentry when deploying releases.
 
 <PlatformSection supported={["apple", "android", "javascript", "flutter", "native", "rust", "python", "react-native", "dotnet", "unity"]}>
 
 ## Release Health
-  
+
+Monitor the [health of releases](/product/releases/health/) by observing user adoption, usage of the application, percentage of [crashes](/product/releases/health/#crash), and [session data](/product/releases/health/#session). Release health will provide insight into the impact of crashes and bugs as it relates to user experience, and reveal trends with each new issue through the release details, graphs, and filters.
```
Comment (imatwawana) [inline src/platforms/common/configuration/releases.mdx:73]: Clarifying where users can find this information 

```suggestion
Monitor the [health of releases](/product/releases/health/) by observing user adoption, usage of the application, percentage of [crashes](/product/releases/health/#crash), and [session data](/product/releases/health/#session). Release health will provide insight into the impact of crashes and bugs as it relates to user experience, and reveal trends with each new issue through the [Release Details](/product/releases/release-details/) graphs and filters.
```

File: src/platforms/common/configuration/releases.mdx [hunk 15]
```diff
@@ -46,26 +46,44 @@ The value can be arbitrary, but we recommend either of these naming strategies:
...
+
+In order to monitor release health the SDK sends session data.
```
Comment (imatwawana) [inline src/platforms/common/configuration/releases.mdx:75]:
```suggestion
In order to monitor release health, the SDK sends session data.
```

**Review by imatwawana — COMMENTED** [review]
This is in a really good place! Thanks for all your work on this. I've made some wording suggestions to improve the flow of in a few places, but otherwise it's looking good to me. I just want to make sure @bruno-garcia gets one more look at the one spot I've called out. I'll do a final review after that, and this will be ready to go!

**Review by bruno-garcia — APPROVED** [review]
:shipit:

File: src/includes/configuration/auto-session-tracking/python.mdx [hunk 16]
```diff
@@ -0,0 +1,17 @@
+We create a session for every request and response cycle and mark the session as:
```
Comment (imatwawana) [inline src/includes/configuration/auto-session-tracking/python.mdx:1]:
```suggestion
We create a session for every request and response cycle, and mark the session as:
```

File: src/includes/configuration/auto-session-tracking/ruby.mdx [hunk 17]
```diff
@@ -0,0 +1,14 @@
+We create a session for every request and response cycle and mark the session as:
```
Comment (imatwawana) [inline src/includes/configuration/auto-session-tracking/ruby.mdx:1]:
```suggestion
We create a session for every request and response cycle, and mark the session as:
```

**Review by imatwawana — APPROVED** [review]
This is good to go! Really great work!

**Comment by antonpirker** [comment]
I have now just removed a short Ruby paragraph, because we want to have the Python documentation update out the door as soon as possible, and Ruby will take a couple of days to be release, so we will the tiny Ruby update later.

**Comment by antonpirker** [comment]
Woohoo, my first docs update done :-)
Thanks @imatwawana and @bruno-garcia for your help!
